### ISSUE TITLE
Split words on symbols

### ISSUE THREAD

**Issue by epage** [bloom42/misspell#3]
Looks like splitting is only done on whitespace.  Splitting on symbols is important for source code so "foo::bar" would be treated as "foo" and "bar" rather than "foobar".

**Comment by z0mbie42** [comment]
totally agree, but I'm not sure what is the most straightforward way to this in rust.

For the moment the I remove non-letter characters in tokens (to catch things like `fn foo(avraible: String)` so `:` will be removed and `avraible` catched

**Comment by epage** [comment]
Looks like #4 is being merged into this.  I've updated the description to include it.

> `fn foo(avraible: String)`

From my understanding of your algorithm, this gets turned into
- `fn`
- `foo(avraible:`
- `String)`

Then removing non-whitespace changes it to
- `fn`
- `fooavraible`
- `String`

Options
- Use regex for splitting on non-whitespace with a separate function to split camel case
- Maybe `nom` could be used to tokenize the file

**Comment by z0mbie42** [comment]
Yes my bad, I mean to say `fn foo(bar: String, avraible: String)` which is an edge case I encountered during development.
